### DISCUSSION TITLE
Documentation Improvements

### DISCUSSION THREAD

**Discussion by dev-embedthis** [sensedeep/dynamodb-onetable#453] [General]
Please post here suggestions or requests for any documentation improvements.

Please cite the topic or page and what is lacking.

**Comment by gregkysenko** [comment]
HI! It was quite hard for me to find how to define an array of objects in schema, I would appreciate having this documented in Readme.

**Reply by mobsense** [reply]
Doc updated at: https://doc.onetable.io/api/table/schemas/modeling/

**Comment by mobsense** [comment]
Hi, thanks for the feedback. I'll make an issue out of this suggestion.

You can create an array property via:

```
Model: {
    list: { type: 'array' }
}
```

If using TypeScript, you can then create a nested schema to type the array elements:

```
Model: {
    list: { type: 'array', item: {
        type: 'object', schema: {
            attribute: {type: 'string'}
        }
    }
}

```

**Comment by DasPoet** [comment]
Hi there, I had an issue today where I tried to perform a `batchGet` operation using a filter that only specified a partition key for a model that was defined in my schema as having a composite key - thus "missing" the sort key.

```ts
export const getModelsByIDs = async (ids: string[]): Promise<Model[]> => {
  const batch = {}

  for (const id of ids) {
    await table
      .getModel('Model')
      .get({ pk: `model-${id}` }, { batch })
  }

  const result: Model[] = await table.batchGet(batch, { parse: true })

  return result.map(parseModel)
}
```

Apparently, OneTable automatically performs a `find` instead of a `get` operation when the primary key is not fully specified. However, this lead to the following error: `[OneTableArgError: Unsupported batch operation "find"]`.

Given that I seemingly just copied [the example given in the documentation](https://doc.onetable.io/api/table/methods/#batchget), this was quite confusing. Perhaps it would be a good idea to include a warning against combining `get` operations using such partial filters with batching.

**Reply by mobsense** [reply]
Yes, OneTable will do a find fallback if the get operation does not provide a full key (PK + SK).

In the example:

```
let batch = {}
await Account.get({id: accountId}, {batch})
await User.get({id: userId}, {batch})
let results = await table.batchGet(batch)
```

It was assumed that the userId enabled the completion of a full key and so would not fallback to find.

I suppose we could disable find-fallback if used with batch and fail the request as you suggest?  I'll add a bug about this.

**Reply by DasPoet** [reply]
Sounds good, thank you very much indeed :)
